src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexWriter.cs

```diff
@@ -135,7 +135,7 @@ public RegexCode RegexCodeFromRegexTree(RegexTree tree)
             CultureInfo culture = (tree.Options & RegexOptions.CultureInvariant) != 0 ? CultureInfo.InvariantCulture : CultureInfo.CurrentCulture;
 
             RegexBoyerMoore? bmPrefix = null;
-            if (prefix.Prefix.Length > 1) // if it's == 1, we're better off using fcPrefix
+            if (prefix.Prefix.Length > 1 && prefix.Prefix.Length <= RegexBoyerMoore.MaxLimit) // if it's <= 1 || > MaxLimit, perf is better using fcPrefix
```

Comment:
Why does RegexCode need an `fcPrefix` if it has a `bmPrefix` ? As far as I can see it could have one or the other. In which case, if this method decides to use Boyer Moore, it can skip calculating `fcPrefix` at all.

Reply:
Seems reasonable.  I'll change it.